satellite/gc/gc_test.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package gc_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/internal/testrand"
+	"storj.io/storj/pkg/encryption"
+	"storj.io/storj/pkg/paths"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite"
+)
+
+// TestGarbageCollection does the following:
+// * Set up a network with one storagenode
+// * Upload two objects
+// * Delete one object from the metainfo service on the satellite
+// * Trigger a bloom filter generation
+// * Check that pieces of the deleted object are deleted on the storagenode
+// * Check that pieces of the kept object are not deleted on the storagenode
+func TestGarbageCollection(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 1, UplinkCount: 1,
+		Reconfigure: testplanet.Reconfigure{
+			Satellite: func(log *zap.Logger, index int, config *satellite.Config) {
+				config.GarbageCollection.FalsePositiveRate = 0.0001
+				config.GarbageCollection.Interval = 0
+			},
+		},
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		satellite := planet.Satellites[0]
+		upl := planet.Uplinks[0]
+		targetNode := planet.StorageNodes[0]
+
+		checker := satellite.Repair.Checker
+		checker.Loop.Stop()
+
+		// Upload two objects
+		testData1 := testrand.Bytes(8 * memory.KiB)
+		testData2 := testrand.Bytes(8 * memory.KiB)
+
+		err := upl.Upload(ctx, satellite, "testbucket", "test/path/1", testData1)
+		require.NoError(t, err)
+		deletedEncPath, pointerToDelete := getPointer(ctx, t, satellite, upl, "testbucket", "test/path/1")
+		var deletedPieceID storj.PieceID
+		for _, p := range pointerToDelete.GetRemote().GetRemotePieces() {
+			if p.NodeId == targetNode.ID() {
+				deletedPieceID = pointerToDelete.GetRemote().RootPieceId.Derive(p.NodeId, p.PieceNum)
+				break
+			}
+		}
+		require.NotNil(t, deletedPieceID)
```

Comment:
`storj.PieceID` is defined as a `[32]byte` array. So it can never be `nil`. `require.NotNil(t, deletedPieceID)` will always be satisfied regardless if a value has been set to `deletedPieceID` inside the loop.

Better use `require.NotZero`.